policies/test.rego

```diff
@@ -0,0 +1,15 @@
+package hacbs.contract.test
+
+# Check if we have any test data provided
+deny = {"msg": msg} {
+	not count(data.test) > 0
+	msg := "No test data provided"
+}
+
+# Check if all tests succeeded 
+deny = {"msg": msg} {
+	results := [r | r := data.test[_].result == "SUCCESS"]
```

Comment:
I'm reading the [Universal Quantification](https://www.openpolicyagent.org/docs/latest/policy-language/#universal-quantification-for-all) doc, and I'm wondering if the current form is "idiomatic" rego.

Actually, I don't see the doc mentioning the `all` function at all in the language spec! :exploding_head:

Reply:
I think it can be, options are either using negation, comprehensions as we do here, or `every`. I've renamed the `results` to make it a bit more readable.

Reply:
I've must have been reading older version of the reference where `all` builtin function was still documented. It was in fact [deprecated](https://github.com/open-policy-agent/opa/issues/2437), so I used `count` instead here.